addons/calendar/models/calendar_event.py

```diff
@@ -1681,6 +1733,19 @@ def _get_date_formats(self):
         lang = get_lang(self.env)
         return (lang.date_format, lang.time_format)
 
+    @api.model
+    def _get_partner_ids_from_vals(self, partner_vals):
+        partner_ids = set()
+        for val in partner_vals:
+            if isinstance(val, (tuple, list)):
```

Comment:
just to be sure i would also check before if the val is already a partner recordset